backend/tests/lib.rs

```diff
@@ -16,43 +32,136 @@ fn test_connected() {
     assert_eq!(response.into_string().unwrap(), "Ok");
 }
 
-#[test]
-fn test_signup() {
+fn test_email() {
     let client = common::get_client().lock();
     let name: String = std::iter::repeat(())
         .map(|()| thread_rng().sample(Alphanumeric))
         .map(char::from)
         .take(16)
         .collect();
+    // set verification code: Request Time 1
+    let response = client
+        .post("/users/email")
+        .json(&json!({
+            "email": format!("{}@mails.tsinghua.edu.cn", name)
+        }))
+        .remote("127.0.0.1:8000".parse().unwrap())
+        .dispatch();
+    assert_eq!(response.status(), Status::Ok);
+    println!("{}", response.into_string().unwrap());
+    std::thread::sleep(std::time::Duration::from_secs(5));
```

Comment:
no need to sleep.

Reply:
Really? I thought we should leave some time for mq consumer to set redis

Reply:
OK， then maybe 1 second is enough